CMakeLists.txt

```diff
@@ -1252,6 +1252,67 @@ set(logical_target_for_zephyr_elf ${logical_target_for_zephyr_elf} PARENT_SCOPE)
 # 2. it can be defined in Kconfig
 set_target_properties(${logical_target_for_zephyr_elf} PROPERTIES OUTPUT_NAME ${KERNEL_NAME})
 
+if (IMAGE STREQUAL mcuboot_ AND CONFIG_MCUBOOT_BUILD_S1_VARIANT)
+# Create second MCUBoot executable for the second slot of the second stage
+# bootloader. This is done inside this file since it is non-trivial to add
+# executable targets outside the root CMakeLists.txt.
```

Comment:
We need to elaborate on, in the comment, why this must be in the root so that we can move it if it happens to become easier in the future.

Reply:
Agree. Also Perhaps try to come up with an upstream solution for this (create executable from anywhere)

Reply:
Extended with 

```
#  ... The problem is that
# most of the variables requried for compiling/linking is only available
# in the scope of this file.